lib/config/config-initializer.js

```diff
@@ -417,16 +429,62 @@ function askInstallModules(modules, packageJsonExists) {
 function promptUser() {
 
     return inquirer.prompt([
+        {
+            type: "list",
+            name: "purpose",
+            message: "How would you like to use ESLint?",
+            default: "problems",
+            choices: [
+                { name: "To check syntax only", value: "syntax" },
+                { name: "To check syntax and find problems", value: "problems" },
+                { name: "To check syntax, find problems, and enforce code style", value: "style" }
+            ]
+        },
+        {
+            type: "list",
+            name: "moduleType",
+            message: "What type of modules does your project use?",
+            default: "esm",
+            choices: [
+                { name: "JavaScript modules", value: "esm" },
+                { name: "CommonJS", value: "commonjs" },
```

Comment:
I suspect the wording here might be confusing, because many users aren't familiar with the term "CommonJS" or the distinction between the two module systems. Someone unfamiliar with modules might think, "well, I don't know what CommonJS means but I'm writing JavaScript, so I'm probably using JavaScript modules", when in fact most people using Node are probably using CommonJS. 

Could we add some additional context to this question? I'm not sure what the best wording would be, but maybe it would look something like this:

> * CommonJS (usually used if your code runs in Node.js)
> * JavaScript modules (usually used if ...)

Alternatively, maybe we could put this question after the browser/node question and change the wording accordingly.

Reply:
I'm not sure that this is any less clear than what we had previously. Prior to this change, there were two questions:

1. Do you use CommonJS?
1. Do you use ES6 modules?

IMHO, it was more confusing before because we were not presenting the two options as related to each other. I worry about adding too many words to these options lest people stop reading.

Reply:
Agreed that asking two questions/adding too many words is not ideal. What about something like this?

> Are you using `import`/`export` syntax?

We would remove the explicit question about CommonJS, and instead enable the `commonjs` env when the user says they're both (a) using Node and (b) not using `import`/`export` syntax.

Reply:
We need to explicitly ask about CommonJS because it can be used in the browser, so we can't logically infer that if they're not using Node.js, then they must not be using CommonJS.

I'm not opposed to changing "JavaScript modules" to "import/export syntax", though, if you feel that makes it clearer.